_unittest/test_26_emit.py

```diff
@@ -996,20 +995,83 @@ def test_interference_script(self):
         # Test with no filtering
         expected_interference_colors = [["white", "green", "yellow"], ["red", "green", "white"]]
         expected_interference_power = [["N/A", -20.0, -20.0], [-20.0, -20.0, "N/A"]]
-        expected_protection_colors = [["white", "yellow", "yellow"], ["yellow", "yellow", "white"]]
-        expected_protection_power = [["N/A", -20.0, -20.0], [-20.0, -20.0, "N/A"]]
 
         interference_colors = []
         interference_power_matrix = []
         protection_colors = []
         protection_power_matrix = []
         interference_colors, interference_power_matrix = interference_type_classification(self.aedtapp)
+        assert interference_colors == expected_interference_colors
+        assert interference_power_matrix == expected_interference_power
+
+    @pytest.mark.skipif(
+        config["desktopVersion"] <= "2023.1" or is_ironpython,
+        reason="Skipped on versions earlier than 2023.2",
+    )
+    def test_protection_level_classification_script(self):
+        self.aedtapp.insert_design("interference2")
+        # place components
+        rad1 = self.aedtapp.modeler.components.create_component("Bluetooth Low Energy (LE)")
+        ant1 = self.aedtapp.modeler.components.create_component("Antenna")
+        if rad1 and ant1:
+            ant1.move_and_connect_to(rad1)
+        rad2 = self.aedtapp.modeler.components.create_component("GPS Receiver")
+        ant2 = self.aedtapp.modeler.components.create_component("Antenna")
+        if rad2 and ant2:
+            ant2.move_and_connect_to(rad2)
+        rad3 = self.aedtapp.modeler.components.create_component("WiFi - 802.11-2012")
+        ant3 = self.aedtapp.modeler.components.create_component("Antenna")
+        if rad3 and ant3:
+            ant3.move_and_connect_to(rad3)
+
+        # Reduce the bluetooth transmit power
+        bands = rad1.bands()
+        for band in bands:
+            band.set_band_power_level(-20)
+
+        # Enable L2 P(Y) GPS Band
+        bands = rad2.bands()
+        for band in bands:
+            for child in band.children:
+                if "L2 P(Y)" in band.node_name:
+                    band.enabled = True
+                else:
+                    band.enabled = False
+
+        # Enable HR-DSSS Ch 1-13 Wifi band
+        bands = rad3.bands()
+        for band in bands:
```

Comment:
This code is now duplicated in 3 tests and is merely used to set up a simple project for the actual testing. We should just save an aedt file with this project configuration and then load it for each of these tests.

Reply:
@maxcapodi78 If you want, myself or @madelinecoleman1 can create the project and make this change.

Reply:
please do it. I'm doing this fix because is causing random crash on build machine. Tests too long can cause issue in build machine because AEDT is not multithread and could run in timeout